iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/LinkData.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.util;
+
+import org.apache.commons.math3.complex.Complex;
+import org.apache.commons.math3.complex.ComplexUtils;
+
+/**
+ * @author José Antonio Marqués <marquesja at aia.es>
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+
+public final class LinkData {
+
+    private LinkData() {
+    }
+
+    static double getFixedX(double x, double epsilonX, boolean applyReactanceCorrection) {
+        return Math.abs(x) < epsilonX && applyReactanceCorrection ? epsilonX : x;
+    }
+
+    static BranchAdmittanceMatrix calculateBranchAdmittance(double r, double x, double ratio1, double alpha1,
+        double ratio2, double alpha2, Complex ysh1, Complex ysh2) {
+
+        Complex a1 = ComplexUtils.polar2Complex(ratio1, alpha1);
+        Complex a2 = ComplexUtils.polar2Complex(ratio2, alpha2);
+
+        Complex ytr;
+        if (r == 0.0 && x == 0.0) {
+            ytr = Complex.ZERO;
+        } else {
+            ytr = new Complex(r, x).reciprocal();
+        }
+
+        BranchAdmittanceMatrix branchAdmittance = new BranchAdmittanceMatrix();
+
+        branchAdmittance.y11 = ytr.add(ysh1).divide(a1.conjugate().multiply(a1));
+        branchAdmittance.y12 = ytr.negate().divide(a1.conjugate().multiply(a2));
+        branchAdmittance.y21 = ytr.negate().divide(a2.conjugate().multiply(a1));
+        branchAdmittance.y22 = ytr.add(ysh2).divide(a2.conjugate().multiply(a2));
+
+        return branchAdmittance;
+    }
+
+    static BranchAdmittanceMatrix kronChain(Complex yFirstConnected11, Complex yFirstConnected12,
```

Comment:
I could be nice if you could send me some elements to better understand this admittance matrix building.

Reply:
I wrote a document explaining how to build the admittance matrix of a branch and kron reduction. We will send it by mail.